packages/react-native/React/Base/RCTUtils.mm

```diff
@@ -383,21 +383,29 @@ CGFloat RCTFontSizeMultiplier(void)
   return mapping[RCTSharedApplication().preferredContentSizeCategory].floatValue;
 }
 
+UIDeviceOrientation RCTDeviceOrientation(void) {
+  return [[UIDevice currentDevice] orientation];
+}
+
 CGSize RCTScreenSize(void)
 {
-  // FIXME: this caches whatever the bounds were when it was first called, and then
-  // doesn't update when the device is rotated. We need to find another thread-
-  // safe way to get the screen size.
-
   static CGSize size;
   static dispatch_once_t onceToken;
   dispatch_once(&onceToken, ^{
     RCTUnsafeExecuteOnMainQueueSync(^{
       size = [UIScreen mainScreen].bounds.size;
     });
   });
-
-  return size;
+  
+  UIDeviceOrientation orientation = RCTDeviceOrientation();
+  
+  BOOL isLandscape = orientation == UIDeviceOrientationLandscapeLeft || orientation == UIDeviceOrientationLandscapeRight;
+  
+  if (isLandscape) {
+    return CGSizeMake(size.height, size.width);
+  } else {
+    return CGSizeMake(size.width, size.height);
+  }
```

Comment:
I'm not really sure whether this block of code works fine in all the situations.

When the app is opened in landscape mode, the dispatch_once will run immediately, and we are going to get a height < width... 🤔 

But we are returning a cg size that says that the width of the CGSize is the height of the screen, so the width of the CGSize is the small side... I'm not sure that's correct...

Reply:
Ah.. yea that's true, what do you think about getting "absolute" value in the RCTUnsafeExecuteOnMainQueueSync so if it's not landscape we reverse it? Or how would you approach it?

Reply:
Yeah, I think it makes sense.
I think that we should check for the orientation also in the `RCTUnsafeExecuteOnMainQueueSync` and then track what's the "long side" and what's the "short side" and then return the CGSize accordingly.

as a `nit`, ObjC has a method [`UIDeviceOrientationIsLandscape(UIDeviceOrientation orientation)`](https://developer.apple.com/documentation/uikit/uideviceorientation/islandscape?language=objc) we should use, instead of checking manually `orientation == UIDeviceOrientationLandscapeLeft || UIDeviceOrientationLandscapeRight`.

If we use it, we can probably do something like:
```
if (UIDeviceOrientationIsLandscape(RCTDeviceOrientation())) {

```
and get rid of the  `UIDeviceOrientation orientation` variable

Reply:
@cipolleschi Thanks for the suggestion, I didn't know about `UIDeviceOrientationIsLandscape`. Pushed a fix!